### PR TITLE
use devcontainer features

### PR DESCRIPTION
**Pull request by MP91**
In the past we used scripts to install several tools. Since the devcontainer features are now officially released, we can make use of them and install the needed tools via them.

### CODE REVIEW

File: Dockerfiles/cpp/.devcontainer/Dockerfile [hunk 1]
```diff
@@ -19,17 +19,10 @@
...
-        curl wget ca-certificates unzip init-system-helpers \
-        bzip2 zip git zsh pigz jq \
-        apt-transport-https dialog libkrb5-3 \
-        libgssapi-krb5-2 libicu[0-9][0-9] liblttng-ust[0-9] \
-        locales sudo ncdu cmake \
-        build-essential build-essential \
-        gcc-aarch64-linux-gnu g++-aarch64-linux-gnu binutils-aarch64-linux-gnu \
-        gcc-x86-64-linux-gnu g++-x86-64-linux-gnu binutils-x86-64-linux-gnu && \
-        apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/*
+    apt-get install -y jq sudo curl wget git cmake build-essential \
```
Comment (doosuu) [inline Dockerfiles/cpp/.devcontainer/Dockerfile:22]: do we need to manually install `git` if it is shipped via devcontainer feature?

File: Dockerfiles/cpp/.devcontainer/Dockerfile [hunk 2]
```diff
@@ -66,20 +59,14 @@ RUN conan remove --builds --src --force '*'
...
-        libgssapi-krb5-2 libicu[0-9][0-9] liblttng-ust[0-9] \
-        libstdc++6 zlib1g locales sudo ncdu man-db strace gnupg2 \
-        # install cpp packages
-        build-essential cmake clang lldb llvm gdb ninja-build cppcheck clang-format-14 clang-tidy-14 && \
-        # clean-up
-        apt-get autoremove -y && apt-get clean -y && rm -rf /var/lib/apt/lists/* && \
-        # redirect format and tidy to the versioned variants
-        update-alternatives --install /usr/bin/clang-format clang-format /usr/bin/clang-format-14 100 && \
-        update-alternatives --install /usr/bin/clang-tidy clang-tidy /usr/bin/clang-tidy-14 100
+    apt-get install -y jq sudo curl wget git \
```
Comment (doosuu) [inline Dockerfiles/cpp/.devcontainer/Dockerfile:62]: See comment about git above

**Review by doosuu — COMMENTED** [review]
Still need to perform a manual test

File: Dockerfiles/cpp/.devcontainer/Dockerfile (hunk 1 above)
Comment (MP91) [reply Dockerfiles/cpp/.devcontainer/Dockerfile:22]: unfortunately yes, due to the python requirements. I tried to shift this installation to be done after the feature stuff but this was not working out really...

Comment (MP91) [reply Dockerfiles/cpp/.devcontainer/Dockerfile:22]: But the feature is needed, to install the latest version of git, ubuntu just has version 2.34...

Comment (doosuu) [reply Dockerfiles/cpp/.devcontainer/Dockerfile:22]: OK

**Comment by doosuu** [comment]
Manual test for Python and C++ shows no problems. LGTM

**Review by doosuu — APPROVED** [review]

**Pull request merged**